### PR TITLE
Improved connection

### PR DESCRIPTION
**Pull request by hitz**
solves issues #1 and #3 

The test for "arbitrary" url isn't very clean.  I think it will error out if you give a bad URL (we bring them up and down all the time, so my test had to use test.encodedcc.org...) I am seeing about setting up a permanent test.demo.encodedcc.org alias for test.encodedcc.org but it's not trivial to get an SSH cert on it.

### CODE REVIEW

File: encode_utils/connection.py [hunk 1]
```diff
@@ -171,10 +174,12 @@ def _set_dcc_mode(self,dcc_mode=False):
     dcc_mode = dcc_mode.lower()
     if dcc_mode not in eu.DCC_MODES:
       #: assume dcc_mode is a valid demo host
-      self.dcc_host = dcc_mode
-      self.dcc_url = 'https://' + dcc_mode + '/'
+      eu.DCC_MODES[dcc_mode] = {
```
Comment (nathankw) [inline encode_utils/connection.py]: Is demo.encodedcc.org the only other possible host, or can there be several demo hosts? If several, I like your way of handling this. If just this extra one host, better we add it directly in __init__.DCC_MODES hash.

File: encode_utils/connection.py [hunk 2]
```diff
@@ -157,9 +157,12 @@ def __init__(self,dcc_mode=None):
     #: to the value of the `DCC_SECRET_KEY` environment variable in the ``_set_api_keys()`` private 
     #: instance method.
     self.secret_key = self._set_api_keys()[1]
-    self.auth = (self.api_key,self.secret_key)
-    if not self.api_key or not self.secret_key:
-      self.log_error("WARNING: API keys not set, all functions have no permission")
+    if self.api_key and self.secret_key:
+      self.auth = (self.api_key,self.secret_key)
+    else:
```
Comment (nathankw) [inline encode_utils/connection.py]: Looks good!

File: encode_utils/tests/test_connection.py [hunk 3]
```diff
@@ -169,5 +174,10 @@ def test_2_make_search_url(self):
     query = "search/?assay_title=ChIP-seq&biosample_type=primary+cell&limit=1&organ_slims=blood&type=Experiment"
     self.assertEquals(res,os.path.join(self.conn.dcc_url,query))
 
+  def test_get(self):
+    res = self.conn.get('experiments/ENCSR502NRF/', frame='object')
```
Comment (nathankw) [inline encode_utils/tests/test_connection.py]: The record you are getting can even be specified more simply ,i.e.
```conn.get('ENCSR502NRF', frame='object')```

File: encode_utils/connection.py (hunk 1 above)
Comment (hitz) [reply encode_utils/connection.py]: All of our machines are either *.demo.encodedcc.org or *.production.encodedcc.org.  Test and production actually get mapped by our nginix proxy (e.g., we have a machine called v67x0-test-master.demo.encodedcc.org that is recieves the https: forwards from test.encodedcc.org.

Production only has the current and most recent production machines, and the 2nd one only for a few days after release.

File: encode_utils/tests/test_connection.py (hunk 3 above)
Comment (hitz) [reply encode_utils/tests/test_connection.py]: for some reason this bounced, I think it wasn't following redirect properly.

File: encode_utils/tests/test_connection.py [hunk 4]
```diff
@@ -18,7 +18,11 @@
 from encode_utils import connection
 from encode_utils import profiles
 
-DATA_DIR = "data"
```
Comment (nathankw) [inline encode_utils/tests/test_connection.py]: I actually have a fix for this data dir issue that I think would be preferred. It sets it in 
``encode_utils/tests/__init__.py``  to ```DATA_DIR = os.path.join(os.path.dirname(__file__),"data")````.

Comment (hitz) [reply encode_utils/tests/test_connection.py]: yeah that was a brutal hack

**Comment by hitz** [comment]
I think I have to revert my commits and the PR should update.

**Comment by hitz** [comment]
I also have a commit that adds .DS_Store (mac spotlight index file) to .gitignore, can I throw that in as well?

**Comment by nathankw** [comment]
Yes

**Comment by hitz** [comment]
UGH don't merge this

**Pull request merged**